src/items.c

```diff
@@ -3,116 +3,188 @@
 
     double min_n = ceil((min_size / store_item_size));
 
-    if (store_item_size * min_n < max_size) return min_n;
+    if (store_item_size * min_n < max_size)
+
+        return min_n;
     return 0;
 }
 
-void items_filter_items(basket_item_s *basket, int basket_size, store_s *store) {
+void items_filter_items(store_s *store)
+{
+    // basket_item_s *basket = __fetch_mock_basket(); // test_get_basket(&basket_size);
 
-    for (int i = 0; i < basket_size; i++) {
+    // TODO Change to read basket from file
+    int basket_size = 0;
+    basket_item_s *basket = test_get_basket(&basket_size);
+
+    store->found_items = malloc(sizeof(found_item_s) * store->items_count);
+    store->missing_items = malloc(sizeof(basket_item_s) * store->items_count);
+
+    for (int i = 0; i < basket_size; i++)
+    {
 
         found_item_s found_item;
         basket_item_s missing_item;
 
-        if (items_find_best_match(basket[i], store, &found_item, &missing_item)) {
+        if (items_find_best_match(basket[i], store, &found_item, &missing_item))
+        {
 
             items_add_item_to_found(found_item, store);
-
-        } else {
+        }
+        else
+        {
 
             items_add_item_to_missing(missing_item, store);
-
         }
     }
 }
 
-void items_add_item_to_found(found_item_s item, store_s *store) {
+void items_add_item_to_found(found_item_s item, store_s *store)
+{
 
     store->found_items[store->found_items_count] = item;
     store->found_items_count++;
-
+    store->found_items_total_price += item.total_price;
 }
 
-void items_add_item_to_missing(basket_item_s item, store_s *store) {
+void items_add_item_to_missing(basket_item_s item, store_s *store)
+{
 
     store->missing_items[store->missing_items_count] = item;
     store->missing_items_count++;
-
 }
 
-found_item_s items_convert_to_found_item(store_item_s input_item, int item_count) {
+found_item_s items_convert_to_found_item(store_item_s input_item, int item_count)
+{
 
     found_item_s tmp = {
-            .product_price = input_item.price,
-            .count = item_count,
-            .size = input_item.size,
-            .unit = input_item.unit,
-            .price_per_unit = input_item.price_per_unit
+        .product_price = input_item.price,
+        .unit = input_item.unit,
     };
 
-    tmp.total_price =  tmp.count * tmp.product_price;
+    if (input_item.unit == UNKNOWN)
+    {
+
+        tmp.count = 1;
+        tmp.total_price = input_item.price;
+    }
+    else
+    {
+
+        tmp.count = item_count;
+        tmp.total_price = input_item.price * item_count;
+    }
+
     strcpy(tmp.name, input_item.name);
 
     return tmp;
+}
+
+int items_compare_item_names(char *name_to_find, char *name_to_search)
+{
+    int buffer_length = strlen(name_to_find) + 3;
+    char buffer[buffer_length];
+
+    snprintf(buffer, buffer_length, "(%s)", name_to_find);
 
+    return parse_try_regex_group(name_to_search, buffer) != NULL;
```

Comment:
Couldn't this just be `strstr` instead? Blindly putting user input into a regex can, in the best case, lead to some very strange behavior.

Reply:
It used to just be: return strstr(name_to_search, name_to_find) != NULL;

Blame says i changed it, but i have absolutely no memory of doing that.